lib/internal/Magento/Framework/App/Action/Action.php

```diff
@@ -92,35 +102,15 @@ public function dispatch(RequestInterface $request)
     {
         $this->_request = $request;
         $profilerKey = 'CONTROLLER_ACTION:' . $request->getFullActionName();
-        $eventParameters = ['controller_action' => $this, 'request' => $request];
-        $this->_eventManager->dispatch('controller_action_predispatch', $eventParameters);
-        $this->_eventManager->dispatch('controller_action_predispatch_' . $request->getRouteName(), $eventParameters);
-        $this->_eventManager->dispatch(
-            'controller_action_predispatch_' . $request->getFullActionName(),
-            $eventParameters
-        );
-        \Magento\Framework\Profiler::start($profilerKey);
+        Profiler::start($profilerKey);
 
         $result = null;
         if ($request->isDispatched() && !$this->_actionFlag->get('', self::FLAG_NO_DISPATCH)) {
-            \Magento\Framework\Profiler::start('action_body');
+            Profiler::start('action_body');
             $result = $this->execute();
-            \Magento\Framework\Profiler::start('postdispatch');
```

Comment:
Looks like those start/stop profiler actions were lost. Are they removed intentionally to simplify the implementation?

Reply:
I'm not sure but looks like something I adopted from previous PRs
Let me create another PR based on that one, that introduces `DispatchProfilerPlugin` on `\Magento\Framework\App\FrontControllerInterface::dispatch`

`beforeDispatch` will trigger `Profiler::start`
`afterDispatch` will trigger `Profiler::stop`

Is that fine for you?

Reply:
OK. Managed to fix that together with Docblock change.